src/components/canrawview/canrawview_p.h

```diff
@@ -103,39 +98,178 @@ class CanRawViewPrivate : public QObject {
         }
     }
 
-private:
-    void writeSortingRules(QJsonObject& json) const
+    bool restoreConfiguration(const QJsonObject& json)
     {
-        json["prevIndex"] = _prevIndex;
-        json["sortIndex"] = _sortIndex;
-        json["currentSortOrder"] = _currentSortOrder;
+        if (columnAdopt(json) == false) {
+            return false;
+        }
+        if (scrollingAdopt(json) == false) {
+            return false;
+        }
+        return true;
     }
 
+private:
     void writeColumnsOrder(QJsonObject& json) const
     {
         int ii = 0;
-        QJsonArray columnList;
+        QJsonArray columnArray;
         for (const auto& column : _columnsOrder) {
             if (_ui.isColumnHidden(ii) == false) {
-                columnList.append(column);
+                QJsonObject columnProper;
+                int vIdx;
+                int width;
+                _ui.getColumnProper(ii, vIdx, width);
+                columnProper["name"] = column;
+                columnProper["vIdx"] = vIdx;
+                columnProper["width"] = width;
+                columnArray.append(columnProper);
             }
             ++ii;
         }
-        json["columns"] = std::move(columnList);
+        json["columns"] = std::move(columnArray);
     }
 
-    void writeViewModel(QJsonArray& jsonArray) const
+    /* In the future for create log file below code can be use it.
+     * Temporary is commented.
+        void writeViewModel(QJsonArray& jsonArray) const
+        {
+            for (auto row = 0; row < _tvModel.rowCount(); ++row) {
+                QJsonArray lineIter;
+                for (auto column = 0; column < _tvModel.columnCount(); ++column) {
+                    if (_ui.isColumnHidden(column) == false) {
+                        auto pp = _tvModel.data(_tvModel.index(row, column));
+                        lineIter.append(std::move(pp.toString()));
+                    }
+                }
+                jsonArray.append(std::move(lineIter));
+            }
+        }
+    */
+
+    bool scrollingAdopt(QJsonObject const& json)
     {
-        for (auto row = 0; row < _tvModel.rowCount(); ++row) {
-            QJsonArray lineIter;
-            for (auto column = 0; column < _tvModel.columnCount(); ++column) {
-                if (_ui.isColumnHidden(column) == false) {
-                    auto pp = _tvModel.data(_tvModel.index(row, column));
-                    lineIter.append(std::move(pp.toString()));
+        auto scrolling = json.find("scrolling");
```

Comment:
Names of properties understood by our objects, like "scrolling" here, or "columns", "name" in other places are hard-coded strings - it would be better to have them at least defined as const std::string.
Also I would propose a list of properties supported by object is defined in it's class, to allow more generic handling of properties, possibly even defining type of property, e.g. as std::map<std::string /*property_name*/, QVariant::Type /*property_type*/> or a map with property value directly stored in it.
Also, as a side note - if we store property values in qt property system, we could make use of notifications on property changes, that could make it easier to reconfigure in real time.

Reply:
I agree. It would be good to have generic parser that will match fields from config files with properties that are supported by a component.